src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.CacheEntryTokens.cs

```diff
@@ -25,13 +25,13 @@ private sealed class CacheEntryTokens
 
             internal void AttachTokens(CacheEntry cacheEntry)
             {
-                if (_expirationTokens != null)
+                if (_expirationTokens is { } expirationTokens)
```

Comment:
Why is an explicit type check more appropriate here? To me that looks something else than a null check and it also adds some fragility/risk here - if in the future the field type were to change (e.g., to a HashSet) then this will either result in a compiler error or potentially even a silent (actual) cast.
The exact type of the field wasn't explicitly visible in this function before also, so why should it matter now?

Reply:
Because `is { }` is a really unusual null check that we avoid in this repo, and "to me looks something else than a null check" :)

You can also just do the easily readable:
```C#
List<IChangeToken>? expirationTokens = _expirationTokens;
if (expirationTokens is not null)
```

Reply:
`is { }` is the best kind of null check 😁
Anyway, I see that some projects in this repo use `var`, so how about `if (_expirationTokens is not null and var expirationTokens)`?
It's a bit sad that some of the cool new C# features are shunned in this repo.

Reply:
> It's a bit sad that some of the cool new C# features are shunned in this repo.

Pattern matching is fine.  But there are now a bazillion ways to do pure null checks; we don't need to inconsistently use them all.

> if (_expirationTokens is not null and var expirationTokens)?

If you want to use var, can't it just be `_expirationTokens is var expirationTokens`?

Reply:
Unfortunately `var` doesn't do a null check and it as been like this from the introduction of `var`.

Reply:
I this OK then?
```cs
var expirationTokens = _expirationTokens;
if (expirationTokens is not null)
```